src/core/server/saved_objects/service/lib/scoped_client_provider.ts

```diff
@@ -135,4 +179,31 @@ export class SavedObjectsClientProvider {
         });
       }, client);
   }
+
+  private getExtensions(
+    request: KibanaRequest,
+    excludedWrappers: string[]
+  ): SavedObjectsExtensions {
+    const isEncryptionExtensionIncluded =
+      !excludedWrappers.includes(ENCRYPTION_EXTENSION_ID) && !!this.encryptionExtensionFactory;
+    const encryptionExtension = isEncryptionExtensionIncluded
+      ? this.encryptionExtensionFactory?.({ typeRegistry: this._typeRegistry, request })
+      : undefined;
+    const isSecurityExtensionIncluded =
+      !excludedWrappers.includes(SECURITY_EXTENSION_ID) && !!this.securityExtensionFactory;
+    const securityExtension = isSecurityExtensionIncluded
+      ? this.securityExtensionFactory?.({ typeRegistry: this._typeRegistry, request })
+      : undefined;
+    const isSpacesExtensionIncluded =
+      !excludedWrappers.includes(SPACES_EXTENSION_ID) && !!this.spacesExtensionFactory;
+    const spacesExtension = isSpacesExtensionIncluded
+      ? this.spacesExtensionFactory?.({ typeRegistry: this._typeRegistry, request })
+      : undefined;
+
+    return {
+      encryptionExtension,
+      securityExtension,
+      spacesExtension,
+    };
+  }
```

Comment:
This mimics the behavior of SOC wrappers in that any extension can be disabled if the extension ID is specified in the `excludedWrappers` option when instantiating a scoped SOC. I didn't want to rename that option as it would be a breaking change. This approach is fully backwards compatible with existing plugin behavior.